metrics/metrics.go

```diff
@@ -26,3 +26,23 @@ func TimeDuration(name string, duration time.Duration) {
 func Gauge(name string, value int64) {
 	metrics.GetOrRegisterGauge(name, metrics.DefaultRegistry).Update(value)
 }
+
+// Counters hold an int64 value that can be incremented and decremented.
+type Counter interface {
+	Clear()
+	Count() int64
+	Dec(int64)
+	Inc(int64)
+	Snapshot() Counter
+}
+
+// GetOrRegisterCounter returns an existing Counter or constructs and registers
+// a new StandardCounter.
+func GetOrRegisterCounter(name string, r Registry) Counter {
+	if nil == r {
+		r = DefaultRegistry
+	}
+	return r.GetOrRegister(name, NewCounter).(Counter)  {
+	metrics.GetOrRegisterCounter(name, metrics.DefaultRegistry).Update(r)
```

Comment:
I think it might be better to abstract away the `go-metrics` library like the other functions are doing, and instead just have a function like `func IncrementCounter(name string, delta int64)` or something like that.